cg/apps/loqus.py

```
@@ -1,34 +1,109 @@
+            stderr=subprocess.STDOUT,
         )
-        self.check_indexes()
+
+        nr_variants = 0
+        # Parse log output to get number of inserted variants
+        for line in output.decode('utf-8').split('\n'):
+            log_message = (line.split('INFO'))[-1].strip()
+            if 'inserted' in log_message or 'Nr of variants in vcf' in log_message:
+                nr_variants = int(log_message.split(':')[-1].strip())
+
         return dict(variants=nr_variants)
 
     def get_case(self, case_id: str) -> dict:
         """Find a case in the database by case id."""
-        return self.db.case.find_one({'case_id': case_id})
+        case_obj = None
+        case_call = copy.deepcopy(self.base_call)
+
+        # For loqusdb v1
+        # loqusdb cases -c is unstable in loqusdb. Here all variants are found
+        # through loqusdb cases (skipping the --case-id option), and the cases
+        # are parsed for the correct case_id
+        case_call.extend(['cases'])
+
+        # For loqusdb v2
+        # case_call.extend(['cases', '-c', case_id, '--to-json'])
+
+        try:
+            output = subprocess.check_output(
+                ' '.join(case_call),
+                shell=True
+            )
+
+        except CalledProcessError:
+            # If case does not exist we will get a non zero exit code and return None
+            return case_obj
+
+        # For loqusdb v1
+        # parse through the output lines to see if case is in loqusdb
+        for line in output.decode('utf-8').split('\n'):
+
+            if line == '':
+                continue
+
+            json_line = line.replace("ObjectId(", '').replace(")", '').replace("'", '"')
+            case = json.loads(json_line)
+            if case_id == case['case_id']:
+                case_obj = case
+                break
+
+        # The output is a list of dictionaries that are case objs
+        # case_obj = json.loads(output.decode('utf-8'))[0]
+
+        return case_obj
```

Could you raise an exception instead or return None?